corehq/apps/ota/views.py

```diff
@@ -174,6 +176,28 @@ def claim(request, domain):
     return HttpResponse(status=200)
 
 
+@location_safe
+@csrf_exempt
+@require_POST
+@check_domain_migration
+def claim_all(request, domain):
+    username = request.POST.get("username")     # username may or may not be fully qualified
+    username = format_username(raw_username(username), domain)
+    user_id = username_to_user_id(username)
```

Comment:
Since the username is coming in as a parameter we should probably also check that the user belongs to the domain, or is that checked elsewhere?

Reply:
Addressed in https://github.com/dimagi/commcare-hq/pull/30039